plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcPageSourceFactory.java

```diff
@@ -437,6 +450,37 @@ else if (transaction.isUpdate()) {
         }
     }
 
+    private static void validateOrcAcidVersion(Optional<AcidInfo> acidInfo, boolean originalFile, OrcReader reader)
+    {
+        if (!originalFile && acidInfo.isPresent() && !acidInfo.get().isOrcAcidVersionValidated()) {
```

Comment:
`!originalFile` is checked by the caller. check it once please.

Reply:
Thanks. I moved whole this if outside and forgot to remove here.